### ISSUE TITLE
[Bug]: LD_LIBRARY_PATH empty

### ISSUE THREAD

**Issue by MrSSvard** [flatpak/flatpak-builder#474]
### Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/flatpak/flatpak/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/flatpak/flatpak-builder/issues) for a bug that matches the one I want to file, without success.

### flatpak-builder version

1.2.2

### Flatpak version

1.12.7

### How to reproduce

I've tried with two separate flatpak builds on two separate computers:

1. Build and install a flatpak (I used --install --user)
2. Enter it with flatpak run --command=bash
3. env | grep LD_LIBRARY_PATH

### Expected Behavior

LD_LIBRARY_PATH should contain at least /usr/lib, /app/lib and and anything set in the manifest.

### Actual Behavior

LD_LIBRARY_PATH is empty.

### Additional Information

I'm running up-to-date Fedora Silverblue 36 (should have the same versions of everything as regular Fedora Workstation 36)

I tried building using both the flatpak-builder in the standard fedora repos and org.flatpak.Builder via flathub (also 1.2.2). Both had the same issue.

This is one of the applications I tried. I successfully built and ran it half a year ago but now it's broken due to the LD_LIBRARY_PATH issue.

I found this issue from a year ago: https://github.com/flatpak/flatpak/issues/4080
But it was only for setuid versions of bwrap. Mine does not have the setuid bit set.

Please ask if there's anything else I can provide.

**Comment by Erick555** [comment]
Yes LD_LIBRARY_PATH is empty by default and it's normal behavior. What's the actual problem you are facing? All relevant paths for dynamic linker are set in `.var/app/<appid>/.ld.so/xxx` which is dynamically generated at runtime.

**Comment by MrSSvard** [comment]
Thanks for the quick reply! Hmm, after testing it some more, I think I've misunderstood some things. 
My main issue was I wasn't sure how I should handle dotnet as a dependency. There's an extension for it but I couldn't get it in the resulting flatpak via add-extensions. 
So then I added a module to download and copy the files to a subfolder in /app/lib but I couldn't find a way to set environment variables in the resulting flatpak. And then I found the old github issue and incorrectly focused on LD_LIBRARY_PATH not being set.

How should I set arbitrary environment variables for the runtime? The docs says there are multiple ways but I didn't find any.
